File: frontend/src/shared/types/inbox.ts
import type { PlatformType } from "./platformConnection";

export type ConversationStatus = "open" | "assigned" | "paused" | "resolved" | "escalated";
export type SenderType = "customer" | "llm_bot" | "human_admin" | "system";
export type MessageDirection = "inbound" | "outbound";
export type MessageDeliveryStatus = "pending" | "queued" | "sent" | "delivered" | "failed";

export type ConversationAssignee = {
  user_id: number;
  full_name: string | null;
  email: string;
};

export type InboxMessage = {
  id: number;
  account_id: number;
  conversation_id: number;
  created_by_user_id: number | null;
  sender_type: SenderType;
  direction: MessageDirection;
  delivery_status: MessageDeliveryStatus;
  sender_name: string | null;
  sender_external_id: string | null;
  external_message_id: string | null;
  content: string;
  error_message: string | null;
  metadata_json: Record<string, unknown>;
  created_at: string;
  updated_at: string;
};

export type ConversationSummary = {
  id: number;
  account_id: number;
  platform_connection_id: number | null;
  platform_type: PlatformType;
  status: ConversationStatus;
  external_thread_id: string | null;
  customer_external_id: string;
  customer_name: string | null;
  customer_avatar_url: string | null;
  customer_phone: string | null;
  customer_email: string | null;
  subject: string | null;
  latest_message_preview: string | null;
  latest_message_at: string | null;
  paused_until: string | null;
  resolved_at: string | null;
  last_inbound_at: string | null;
  unread_count: number;
  assigned_to: ConversationAssignee | null;
  created_at: string;
  updated_at: string;
};

export type ConversationDetail = ConversationSummary & {
  messages_total: number;
  messages: InboxMessage[];
};

export type ConversationListResponse = {
  items: ConversationSummary[];
  total: number;
  page: number;
  page_size: number;
};

export type MessageListResponse = {
  items: InboxMessage[];
  total: number;
  page: number;
  page_size: number;
};

File: backend/app/schemas/inbox.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import (
    ConversationStatus,
    MessageDeliveryStatus,
    MessageDirection,
    PlatformType,
    SenderType,
)
from app.schemas.common import ORMModel


class ConversationAssigneeResponse(BaseModel):
    user_id: int
    full_name: str | None
    email: str


class ConversationSummaryResponse(ORMModel):
    id: int
    account_id: int
    platform_connection_id: int | None
    platform_type: PlatformType
    status: ConversationStatus
    external_thread_id: str | None
    customer_external_id: str
    customer_name: str | None
    customer_avatar_url: str | None
    customer_phone: str | None
    customer_email: str | None
    subject: str | None
    latest_message_preview: str | None
    latest_message_at: datetime | None
    paused_until: datetime | None
    resolved_at: datetime | None
    last_inbound_at: datetime | None
    unread_count: int = 0
    assigned_to: ConversationAssigneeResponse | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class MessageResponse(ORMModel):
    id: int
    account_id: int
    conversation_id: int
    created_by_user_id: int | None
    sender_type: SenderType
    direction: MessageDirection
    delivery_status: MessageDeliveryStatus
    sender_name: str | None
    sender_external_id: str | None
    external_message_id: str | None
    content: str
    error_message: str | None
    metadata_json: dict[str, Any]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ConversationDetailResponse(ConversationSummaryResponse):
    messages_total: int = 0
    messages: list[MessageResponse] = Field(default_factory=list)


class ConversationListResponse(BaseModel):
    items: list[ConversationSummaryResponse]
    total: int
    page: int
    page_size: int


class MessageListResponse(BaseModel):
    items: list[MessageResponse]
    total: int
    page: int
    page_size: int


class ConversationAssignRequest(BaseModel):
    assignee_user_id: int | None = None


class ConversationStatusUpdateRequest(BaseModel):
    status: ConversationStatus
    paused_until: datetime | None = None


class SendReplyRequest(BaseModel):
    content: str = Field(..., min_length=1, max_length=5000)
    sender_type: SenderType = SenderType.HUMAN_ADMIN
